config/manager/manager.yaml

```diff
@@ -36,35 +36,10 @@ spec:
       labels:
         control-plane: controller-manager
     spec:
-      # TODO(user): Uncomment the following code to configure the nodeAffinity expression
-      # according to the platforms which are supported by your solution. 
-      # It is considered best practice to support multiple architectures. You can
-      # build your manager image using the makefile target docker-buildx.
-      # affinity:
-      #   nodeAffinity:
-      #     requiredDuringSchedulingIgnoredDuringExecution:
-      #       nodeSelectorTerms:
-      #         - matchExpressions:
-      #           - key: kubernetes.io/arch
-      #             operator: In
-      #             values:
-      #               - amd64
-      #               - arm64
-      #               - ppc64le
-      #               - s390x
-      #           - key: kubernetes.io/os
-      #             operator: In
-      #             values:
-      #               - linux
       securityContext:
         runAsNonRoot: true
```

Comment:
IIRC, `runAsUser` was required as well.
Can you please verify the Deployment can be deployed on a restricted NS?

`kubectl label --dry-run=server --overwrite ns ns-name pod-security.kubernetes.io/enforce=restricted`

Reply:
Also, I cannot add the comment on the relevant line so writing here- the name "system" for the namespace is problematic. Please use the name suggested in the design doc, or another more specific name.

Reply:
> IIRC, `runAsUser` was required as well. Can you please verify the Deployment can be deployed on a restricted NS?
> 
> `kubectl label --dry-run=server --overwrite ns ns-name pod-security.kubernetes.io/enforce=restricted`

Yes, it works, no warning /error raised


> Also, I cannot add the comment on the relevant line so writing here- the name "system" for the namespace is problematic. Please use the name suggested in the design doc, or another more specific name.


Regarding the namespce, it eventually change by Kustomize  https://github.com/AlonaKaplan/selfserviceoverlay/blob/main/config/default/kustomization.yaml#L2
`make deploy` result is creating the mentioned namespace and all related objects inside it

Reply:
> You can achieve that by specifying runAsUserwith UID that is not root (not zero)

yes, that's what kubevirt and kubesecondarydns do. I wonder how you can created the Deployment in a restricted ns without it. I remember that in the past I"ve got a warning the `runAsUserwith` is missing.

Reply:
Its possible because `runAsNonRoot` is spesified.

Reply:
But you have to specify the non-root user id.

Reply:
RunAsNonRoot make sure the container wont run with UID: 0,  Kubernetes will validate that the container image is configured to run as a non-root user at runtime.
RunAsUser is more granular in the sense that it enable specifying the UID.

According to the restricted pod security policy doc, k8s validates that `RunAsNonRoot` set and `true` (mandatory), and/or RunAsUser is not zero (seem optional):
See `Running as Non-root` and `Running as Non-root user` in the table:
https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted